osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/MainCirclePiece.cs

```diff
@@ -51,17 +51,22 @@ private void load()
             var drawableOsuObject = (DrawableOsuHitObject)drawableObject;
 
             state.BindTo(drawableObject.State);
-            state.BindValueChanged(updateState, true);
-
             accentColour.BindTo(drawableObject.AccentColour);
+            indexInCurrentCombo.BindTo(drawableOsuObject.IndexInCurrentComboBindable);
+        }
+
+        protected override void LoadComplete()
```

Comment:
This change should be pretty self-explanatory - `HitObject` may not be set during `load()`. These changes match what's done in `LegacyCirclePiece` and `LegacySpinner`, etc.